staging/src/k8s.io/cli-runtime/pkg/printers/json.go

```diff
@@ -140,3 +151,13 @@ func (p *YAMLPrinter) PrintObj(obj runtime.Object, w io.Writer) error {
 	_, err = fmt.Fprint(w, string(output))
 	return err
 }
+
+func stripManagedFields(obj runtime.Object) runtime.Object {
```

Comment:
is it possible to build a "ManagedFieldsStripper" Printer, that will encapsulate another stripper but remove the ManagedFields from the object first? I think that'd be a slightly nicer design!

Reply:
Yeah I think adding a new printer is better and I have added an `OmitManagedFields` printer similar to your proposal.